### PR TITLE
[ST] Clean up SpecificIsolatedST - triage

### PR DESCRIPTION
**Pull request by jankalinic**
### Type of change

- Refactoring

### Description

This PR consists of modifications improving structure of our STs. Removing unnecessary STs and moving wrongly put ones to their respectively correct suites.

### Checklist

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Comment by jankalinic** [comment]
@strimzi-ci run tests --profile=regression --install-type=bundle --ocp-version=4.13 --env=ST_KAFKA_VERSION=3.4.0

**Comment by jankalinic** [comment]
@strimzi-ci run tests --install-type=bundle --ocp-version=4.13 --testcase=KafkaST#testDeployUnsupportedKafka

File: systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java [hunk 1]
```diff
@@ -945,6 +946,29 @@ void testReadOnlyRootFileSystem(ExtensionContext extensionContext) {
...
+        String nonExistingVersionMessage = "Unsupported Kafka.spec.kafka.version: " + nonExistingVersion + ". Supported versions are:.*";
+        String clusterName = mapWithClusterNames.get(extensionContext.getDisplayName());
+
+        resourceManager.createResource(extensionContext, false, KafkaTemplates.kafkaEphemeral(clusterName, 1, 1)
+                .editSpec()
+                .editKafka()
+                .withVersion(nonExistingVersion)
+                .endKafka()
+                .endSpec()
+                .build());
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java:962]: indent

**Review by im-konge — APPROVED** [review]

File: systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java (hunk 1 above)
Comment (jankalinic) [reply systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java:962]: Thanks, good catch!

**Review by see-quick — APPROVED** [review]

File: systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java [hunk 2]
```diff
@@ -945,6 +946,29 @@ void testReadOnlyRootFileSystem(ExtensionContext extensionContext) {
...
     }
 
+    @ParallelNamespaceTest
+    void testDeployUnsupportedKafka(ExtensionContext extensionContext) {
+        final TestStorage testStorage = new TestStorage(extensionContext);
+        String nonExistingVersion = "6.6.6";
+        String nonExistingVersionMessage = "Unsupported Kafka.spec.kafka.version: " + nonExistingVersion + ". Supported versions are:.*";
+        String clusterName = mapWithClusterNames.get(extensionContext.getDisplayName());
+
+        resourceManager.createResource(extensionContext, false, KafkaTemplates.kafkaEphemeral(clusterName, 1, 1)
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java:956]: shouldn't you pass namespaceName here as well?

Comment (jankalinic) [reply systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java:956]: This PR is only for kafkaTopic related changes, i can do another one with other resources.

**Pull request merged**